specification/resources/resource-manager/Microsoft.Resources/stable/2025-03-01/resources.json

```diff
@@ -0,0 +1,7094 @@
+        "propertyChangeType": {
+          "type": "string",
+          "description": "The type of property change.",
+          "enum": [
+            "Create",
+            "Delete",
+            "Modify",
+            "Array",
+            "NoEffect"
+          ],
+          "x-ms-enum": {
+            "name": "PropertyChangeType",
+            "modelAsString": false,
+            "values": [
+              {
+                "value": "Create",
+                "description": "The property does not exist in the current state but is present in the desired state. The property will be created when the deployment is executed."
+              },
+              {
+                "value": "Delete",
+                "description": "The property exists in the current state and is missing from the desired state. It will be deleted when the deployment is executed."
+              },
+              {
+                "value": "Modify",
+                "description": "The property exists in both current and desired state and is different. The value of the property will change when the deployment is executed."
+              },
+              {
+                "value": "Array",
+                "description": "The property is an array and contains nested changes."
+              },
+              {
+                "value": "NoEffect",
+                "description": "The property will not be set or updated."
+              }
+            ]
+          }
+        },
+        "before": {
+          "type": "object",
+          "description": "The value of the property before the deployment is executed."
+        },
+        "after": {
+          "type": "object",
+          "description": "The value of the property after the deployment is executed."
+        },
+        "children": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/WhatIfPropertyChange"
+          },
+          "x-ms-identifiers": [
+            "path"
+          ],
+          "description": "Nested property changes."
+        }
+      },
+      "description": "The predicted change to the resource property."
+    },
+    "WhatIfChange": {
+      "description": "Information about a single resource change predicted by What-If operation.",
+      "required": [
+        "changeType"
+      ],
+      "properties": {
+        "resourceId": {
+          "type": "string",
+          "description": "Resource ID"
+        },
+        "deploymentId": {
+          "type": "string",
+          "description": "The resource id of the Deployment responsible for this change."
+        },
+        "symbolicName": {
+          "type": "string",
+          "description": "The symbolic name of the resource responsible for this change."
+        },
+        "identifiers": {
+          "type": "object",
+          "description": "A subset of properties that uniquely identify a Bicep extensible resource because it lacks a resource id like an Azure resource has."
+        },
```

Comment:
@anthony-c-martin Could you help add an extension property here? For Stacks, Kyle introduced some design changes where extensions are now responsible for generating the configId property based on the extension configs object. As a result, configId is no longer part of identifiers, meaning users will need to combine identifiers and configId to uniquely identify extensible resources. David's Swagger PR predates this design change.

Technically, we could incorporate the changes in both Before and After, but that would introduce duplicates. It might make more sense to add an extension property at the `WhatIfChange` level instead:

```JSON
{
  "extension": {
    "alias": "k8s",
    "name": "Kubernetes",
    "version": "1.0.0",
    "configId": "..."
  }
}
```

Reply:
I'm not sure I understand exactly which changes you're suggesting to make - would you or Kyle be able to push to this branch, or share a diff with me?